src/Microsoft.ML.Data/DataLoadSave/CompositeDataLoader.cs

```diff
@@ -15,7 +16,7 @@ namespace Microsoft.ML.Data
     /// This class represents a data loader that applies a transformer chain after loading.
     /// It also has methods to save itself to a repository.
     /// </summary>
-    public sealed class CompositeDataLoader<TSource, TLastTransformer> : IDataLoader<TSource>
+    public sealed class CompositeDataLoader<TSource, TLastTransformer> : IDataLoader<TSource>, IDisposable
```

Comment:
📝 By making the class disposable when `IDataLoader<TSource>` is not disposable, it will be easy for a consumer to fail to dispose of the object.

Reply:
We can't change interfaces now that they are shipped. Right now, when customers discover they have a memory leak, we don't have a solution. With this method at least we can tell them to add a Dispose call. 

Is there a better approach we can consider?


---
In reply to: [396132696](https://github.com/dotnet/machinelearning/pull/4939#discussion_r396132696) [](ancestors = 396132696)

Reply:
Every case that I reviewed locally required breaking API changes. Default interface methods might be an option for .NET Core targets.

Reply:
Is the component GA? Or can we still make a breaking change?

Reply:
All public interfaces are frozen. This would be one of those. 

---
In reply to: [396753025](https://github.com/dotnet/machinelearning/pull/4939#discussion_r396753025) [](ancestors = 396753025)

Reply:
> All public interfaces are frozen. This would be one of those.

There may be a difference between interfaces released as GA and the preview releases. I'm not sure which one this falls under.

/cc @eerhardt

Reply:
IDataLoader is a core interface. It is part of the GA set. 

---
In reply to: [396760924](https://github.com/dotnet/machinelearning/pull/4939#discussion_r396760924) [](ancestors = 396760924)